test/acme_server/nonce_test.exs

```diff
@@ -0,0 +1,20 @@
+defmodule AcmeServer.NonceTest do
+  use ExUnit.Case, async: true
+
+  test "new" do
+    first_nonce = AcmeServer.Nonce.new()
+    nonce = AcmeServer.Nonce.new()
+    assert nonce == first_nonce + 1
```

Comment:
This assertion tests an implementation detail. I don't think we should care that the next nonce is incremented by one. In fact, with concurrent tests, this might fail, for example if another test running in parallel generates nonce after `first_nonce` is generated, but before `another_nonce` is generated.

I think a better assertion would be something like `assert another_nonce != first_nonce`.

Also, in case you want to give it a try, I think that this is a nice candidate for testing with [stream_data](https://github.com/whatyouhide/stream_data). Instead of generating just two nonces, you could generate a list of nonces, and then in the property verify that no two nonces are the same. In another property, you could test that every generated nonce can be verified once. Finally, in a third property, you could test that a nonce can only be verified once (i.e. that a repeated verification of the same nonce fails).